langchain/document_loaders/sitemap.py

```diff
@@ -76,6 +87,16 @@ def load(self) -> List[Document]:
 
         els = self.parse_sitemap(soup)
 
+        if self.blocksize is not None and self.blocknum is not None:
+            elblocks = list(_batch_block(els, self.blocksize))
+            blockcount = len(elblocks)
+            if blockcount - 1 < self.blocknum:
```

Comment:
yeah, so there are requirements of these two values. They could be invalid in some cases. I think there are several places where we can place this check. Intuitively, the earlier the better, which points to the class initiation process.

In `pydantic`, which we’re using here, it has a concept of validator, which can provide certain assertions during the process of initiation. With a validator, we can validator both `blocksize` and `blockcount` at the same time and raise exceptions or auto-fix? or silently fallback to default.

reference and examples in https://docs.pydantic.dev/latest/usage/validators/

Reply:
how should i validate the blockcount any earlier then the calculation of the blockcount on line 92?

the blockcount is the number of blocks when all scraped URLs are scraped from the sitemap and then splitup into block of the size given in `self.blocksize`

Reply:
yeah, you’re right for the `blockcount` which is dynamic. You’re more familiar with the biz context here, there might be other numeric check which we can have. From my brief context, for the type check, we can do it earlier. 

I’m thinking if I’m a first time user and I want to call use this class, what common mistakes I could make? One common mistake I might make would be passing in only one of the two params and assuming I can use default value for the other, which, however, is None instead of 0 or another usable number. I think `None` makes sense here because we need the user enable it explicitly. Following this idea, might be good to raise the error early to the user before running the real logics.

Reply:
it could make sense to make the blocknum a 0 by default so if only a blocksize is given the first block will be scraped
also a validator that the blocknum should be larger then 0
and i could think of a validator that the blocksize should be larger then 0

but a check if a blockcount is valid is only possible once all the urls are in the list

Reply:
ah I see. Sorry I made a typo above. I was referring to `blocksize` and `blocknum` above in my first comment… I’m so sorry! `blockcount` is not even a class field, `validator` won’t be able to support anyhow…

My bad! What you answered here definitely makes sense. :)

Reply:
i added simple checks in the __init__ as this class does not use `pydantic` yet
i also run lint and formatter on the 2 files

Reply:
thank you for the updates! appreciate it!